File: app/src/main/kotlin/com/spacex/ui/welcome/WelcomeModule.java
package com.spacex.ui.welcome;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class WelcomeModule {

  @Provides
  static WelcomeContract.Presenter providePresenter(WelcomeContract.View view) {
    return new WelcomePresenter(view);
  }

  @Binds
  abstract WelcomeContract.View bindView(WelcomeActivity welcomeActivity);
}

File: app/src/main/kotlin/com/spacex/ui/welcome/WelcomeModule.kt
package com.spacex.ui.welcome

import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
internal abstract class WelcomeModule {

    @Binds
    abstract fun bindView(welcomeActivity: WelcomeActivity): WelcomeContract.View

    @Module
    companion object {
        @JvmStatic
        @Provides
        fun providePresenter(view: WelcomeContract.View): WelcomeContract.Presenter = WelcomePresenter(view)
    }
}